app/models/group.rb

```diff
@@ -0,0 +1,30 @@
+# == Schema Information
+#
+# Table name: groups
+#
+#  id         :bigint           not null, primary key
+#  name       :string           not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#  company_id :bigint           not null
+#
+# Indexes
+#
+#  index_groups_on_company_id  (company_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (company_id => companies.id)
+#
+class Group < ApplicationRecord
+  belongs_to :company
+  has_and_belongs_to_many :users
+
+  validates :name, presence: true,
+                   uniqueness: { scope: :company },
+                   length: { maximum: 100 }
+
+  def user_count
+    users.size
+  end
```

Comment:
If a performance improvement is needed, this can be improved with a counter-cache like approach. Although counter caches are not really suitable for HABTM relations, we can achieve this through a custom function and a... err.... callback.